crates/red_knot/src/types.rs

```diff
@@ -51,13 +59,13 @@ impl TypeStore {
         self.modules.remove(file_id);
     }
 
-    pub fn cache_symbol_type(&mut self, file_id: FileId, symbol_id: SymbolId, ty: Type) {
```

Comment:
This change should go along with a TODO in this impl for proper synchronization. (I had put the TODO out on `eval_symbols` (now `infer_symbol_types`) since I hadn't changed its db ref to non-mut yet.)